packages/zql/src/builder/builder.test.ts

```diff
@@ -1938,6 +1938,136 @@ test('not exists self join', () => {
         },
         "type": "remove",
       },
+      {
```

Comment:
The big increase in pushes here is from us splitting edit into remove and add for all connections.  The end result is the same (a bunch of rows are added, and then removed).